### ISSUE TITLE
Improve mouse interactions in controls

### ISSUE THREAD

**Issue by kirsan31** [ScottPlot/ScottPlot#999]
# Tasks
_This section was added by @swharden_
- [x] trigger mouse enter and leave events in user controls (#1040)
- [x] provide options for automatic axis limits to ignore axis lines and spans (#1083)
- [x] implement an axis crosshair label feature (#1093)

# Original Question

### Your Environment
- ScottPlot Version: 4.1.12 beta
- Operating System: Win10 20H2
- Is it a GUI app? What platform? WinForms
- .Net:  `net472` and `net5.0-windows` was checked.

Simple Mdi Winforms app. Child form with only formsPlot1 on it. `MouseLeave` and `MouseEnter` not working, `MouseMove` is working.

This is critical to implement some info on `MouseMove` (to hide it after we leave the plot).

P.s. Related question:
How to properly implement axes cursor on `MouseMove` with ScottPlot?
![image](https://user-images.githubusercontent.com/17767561/116544741-32a59400-a8f8-11eb-90bc-fe43cdb62cfe.png)

I was tring something like this:
```cs
private void formsPlot1_MouseMove(object sender, MouseEventArgs e)
{
    (double x, double y) = formsPlot1.GetMouseCoordinates();
    _HLine.Y = y;
    _HLine.IsVisible = true;
    _VLine.X = x;
    _VLine.IsVisible = true;
    formsPlot1.Render(skipIfCurrentlyRendering: true);
}
```
And it's worked BUT. Because of `_HLine` and `_VLine` are plots, ScottPlot autozoom and autoaxes always trying to consider their values. And this lead to whole wrong behavior. For example:

https://user-images.githubusercontent.com/17767561/116545566-47cef280-a8f9-11eb-9c29-ee21ead7fceb.mp4

Cursors must be completely excluded from plotting logic.

Also, is it possible to display Cursor coordinates on the axes (red circles on screen above)?

**Comment by StendProg** [comment]
Hi @kirsan31,

1. #892 is related issue. But it doesn't solve your specific problem.
But it doesn't solve your specific problem.
I think the whole point is that the `PictureBox` rises all events correctly, including `MouseLeave` and `MouseEnter`, but they are not redirected to the Control in any way. And `PictureBox` is in private. 
The solution would be either to provide the user with access to the `PictureBox`, or to redirect all his events to the control (the problem is that there are too many of them).

2. The problem of calculating limits for `Lines` has already been discussed, you can try the solution described in https://github.com/ScottPlot/ScottPlot/issues/643#issuecomment-738117885 
and adopted to 4.1 https://github.com/ScottPlot/ScottPlot/issues/957#issuecomment-816756522

3. > Also, is it possible to display Cursor coordinates on the axes (red circles on screen above)?

At the moment this is impossible, but the idea is good, of course there will be problems with circles, but we can make something like notches directly on the axes.

**Comment by kirsan31** [comment]
@StendProg 
Thank you for your answer.
1. I think we need redirect all mouse events at least?
2. Nice work around, thanks.
3. 
   > of course there will be problems with circles.  

   No no, this circles there only to point the place where coordinates must appear.

Overall, for now 1. is the main bloker to implement cursors on ScottPlot.

**Comment by breakwinz** [comment]
I am not sure if this is related. I am using same version and setup. I just use the "Enter" and "Leave" events and it works great. I am not using Mouse_Enter and Mouse_Leave. However as far as I know. Enter and Leave do the same thing as the mouse events

**Comment by swharden** [comment]
## MouseEnter and MouseLeave

> MouseLeave and MouseEnter not working

These probably are worth forwarding. I'll try to implement these in the next few days, with a demo. Hiding the crosshairs when the mouse exits the plot seems like a good demo.

Keep in mind that for extreme custom mouse interaction is probably just for the programmer to make their own control and implement the features they want. The controls aren't very complicated, and it's much easier to start with the source code and modify it to meet your needs than to try to adapt the standard controls to meet every use case. https://github.com/ScottPlot/ScottPlot/blob/master/src/controls/ScottPlot.WinForms/FormsPlot.cs

## Option to ignore axis lines when calling AxisAuto()

The video demonstrates this problem well. Thanks for sharing it! As @StendProg pointed out this question has been raised before. I can add a public field to axis lines so they are ignored when axis limits are determined automatically, which may be a simpler solution.

## Labeling crosshair coordinates on the axes

Right now this is not possible, but it should be easy to create a special plot type that does this. I'll put this idea on the triage list (#1028). This type of behavior would be especially useful for financial charts, similar to https://www.tradingview.com/chart

![image](https://user-images.githubusercontent.com/4165489/118429041-c7511580-b69e-11eb-81ab-6b759a2a35ed.png)

**Comment by swharden** [comment]
All 3 controls now forward enter/leave events and the mouse tracker demos were updated to demonstrate how to use these events. It seems to work no matter what is in focus. This functionality will be in the next release on NuGet, hopefully in a few days 👍 

@bclehmann I noticed in the Avalonia control that `OnMouseMove` is commented-out. Should this be uncommented or deleted?

https://github.com/ScottPlot/ScottPlot/blob/03002e94c0718f5e8a9b3ea691572c399f9fb239/src/controls/ScottPlot.Avalonia/AvaPlot.axaml.cs#L94

I'll leave this issue open while I work on the remaining tasks
